Assets/PatchKit/Securing by the license key/SecuringByLicenseKey.cs

```diff
@@ -1,204 +1,312 @@
+        private void Exit()
+        {
+            switch (Option)
+            {
+                case Options.TimeStopStart:
+                    Time.timeScale = 1;
+                    break;
+                case Options.EnablesSelectedObjects:
+                    Objects.ForEach(o=> o.SetActive(true)); 
+                    break;
+                case Options.LoadNewScene:
+                    SceneManager.LoadScene(SceneLoad.name);
+                    break;
+                case Options.CallbackEntryAndExit:
+                    OnExit();
+                    break;
+            } 
+        }
+    
+        void Start()
+        {
+#if UNITY_EDITOR
+           if(NotExecuteInEditor)
+                return;
+           PrefabUtility.RecordPrefabInstancePropertyModifications(this);
+#endif
+            if (_isFirstStart)
+            {
+                _isFirstStart = false;
+                _guiStyle = new GUIStyle();
+                _guiStyle.alignment = TextAnchor.UpperCenter;
+                _guiStyle.wordWrap = true;
 
-            _appDataPath = MakeAppDataPathAbsolute();
-            _filePath = Path.Combine(_appDataPath, AppDataFileName);
+                Entry();
+                var checkNetwork = GetResponse("", "", _host);
+                if (checkNetwork == null)
+                {
+                    _labelMessage = "No internet connection";
+                    _contentMessage = "Your are not connected to internet. Please check your internet connection!";
+                    _show = true;
+                    return;
+                }
+                
+                if(string.IsNullOrEmpty(AppSecret))
+                {
+                    _labelMessage = "App secret Application is empty";
+                    _contentMessage = "App secret is empty. Please contact with your developer!";
+                    _show = true;
+                    return;
+                }
+                _appDataPath = MakeAppDataPathAbsolute();
+                _filePath = Path.Combine(_appDataPath, AppDataFileName);
 
-            Debug.Log(_filePath);
-            if (File.Exists(_filePath))
-            {
-                if (TryLoadDataFromFile())
+                Debug.Log(_filePath);
+                if (File.Exists(_filePath))
                 {
-                    var keyInfo = GetKeyInfo();
-                    if (keyInfo != null && keyInfo.StatusCode == HttpStatusCode.OK)
+                    if (TryLoadDataFromFile())
                     {
-                        Time.timeScale = 1;
-                        return;
+                        var keyInfo = GetKeyInfo(_tmpKey);
+                        if (keyInfo != null && keyInfo.StatusCode == HttpStatusCode.OK)
+                        {
+                            Exit();
+                            return;
+                        }
+                        else
+                        {
+                            _labelMessage = "License key is not found";
```

Comment:
There more HTTP codes that this endpoint could return, pointing to different situations:
- 404 (license key is not valid)
- 410 (license key is blocked)
- any other HTTP code should be treated as a connection issue

Reply:
done